### PR TITLE
schedule: fix scatter range

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
There are some mistakes created by the previous PR which causes the scatter range scheduler cannot work.

### What is changed and how it works?
This PR fixes it to make it work like before.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release notes

### CODE REVIEW

**Comment by nolouch** [comment]
Can we improve the metrics? now, it shares the same metrics with others.

**Comment by rleungx** [comment]
@nolouch The way I used is not elegant. I am not sure if there is a better to implement it.

File: server/schedulers/balance_leader.go [hunk 1]
```diff
@@ -34,30 +35,42 @@ const balanceLeaderRetryLimit = 10
...
 	base := newBaseScheduler(opController)
+
 	s := &balanceLeaderScheduler{
 		baseScheduler: base,
 		selector:      schedule.NewBalanceSelector(core.LeaderKind, filters),
 		taintStores:   taintStores,
 		opController:  opController,
 	}
+	if len(name) != 0 {
+		s.leaderCounter = scatterRangeLeaderCounter
```
Comment (nolouch) [inline server/schedulers/balance_leader.go]: Can we use a option to set up for this?

**Pull request merged**